### PR TITLE
[INLONG-10823][SDK] Transform SQL support Reverse function

### LINKED ISSUE #10823: [Feature][SDK] Transform SQL support Reverse function
### Description

- parent  #10796
- REVERSE(str)
Returns the string str with the order of the characters reversed.
- Such as: REVERSE("abc") = "cba"

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ying-hua**
Fixes #10823

### Motivation

Add one transform function class: ReverseFunction and its unit test.
- REVERSE(str)
Returns the string str with the order of the characters reversed.
- Such as: REVERSE("abc") = "cba"
### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     * @param context
+     * @return
+     */
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        String str = OperatorTools.parseString(stringObj);
+        String reversed = new StringBuilder(str).reverse().toString();
+        return reversed;
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:52]:
```suggestion
        return new StringBuilder(str).reverse().toString();
```

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Function;
+
+public class ReverseFunction implements ValueParser {
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:27]: add descriptions for this functions

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    private ValueParser stringParser;
+
+    /**
+     * Constructor
+     * @param expr
+     */
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:34]: add some meaningful docs or just remove it

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    /**
+     * parse
+     * @param sourceData
+     * @param rowIndex
+     * @param context
+     * @return
+     */
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:45]: ditto

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java (hunk 2 above)
Comment (ying-hua) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:32]: OK. updated.

**Review by vernedeng — APPROVED** [review]

**Comment by aloyszhang** [comment]
@ying-hua please resolve the conflict files.

**Comment by ying-hua** [comment]
> @ying-hua please resolve the conflict files.

OK.

**Comment by ying-hua** [comment]
I will submit another pull request to resole the conflict.

**Pull request closed without merging**